### PR TITLE
new style normalize function

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR implements new style `normalize` function.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Confirmed 10000 repeats.

File: chainer/functions/normalization/l2_normalization.py [hunk 1]
```diff
@@ -22,22 +23,25 @@ def check_type_forward(self, in_types):
...
-        x = inputs[0]
-        gy = gy[0]
-        xp = cuda.get_array_module(x)
+    def backward(self, indexes, grad_outputs):
+        x, = self.get_retained_inputs()
+        gy, = grad_outputs
 
-        norm = xp.linalg.norm(x, axis=self.axis) + self.eps
-        norm = xp.expand_dims(norm, self.axis)
+        norm = chainer.functions.sqrt(chainer.functions.sum(
```
Comment (niboshi) [inline chainer/functions/normalization/l2_normalization.py]: How about assigning `F = chainer.functions` before these calculations? The code could be cleaner.

Comment (kmaehashi) [reply chainer/functions/normalization/l2_normalization.py]: Thank you, fixed as suggested.

**Comment by niboshi** [comment]
Thanks!
(jenkins has passed. Awaiting AppVeyor)

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**